internal/querynodev2/segments/result.go

```diff
@@ -551,23 +557,20 @@ func MergeSegcoreRetrieveResults(ctx context.Context, retrieveResults []*segcore
 
 		for _, r := range segmentResults {
 			if len(r.GetFieldsData()) != 0 {
-				ret.FieldsData = typeutil.PrepareResultFieldData(r.GetFieldsData(), int64(len(selected)))
+				ret.FieldsData = typeutil.PrepareResultFieldData(r.GetFieldsData(), int64(len(selections)))
 				break
 			}
 		}
 
 		_, span3 := otel.Tracer(typeutil.QueryNodeRole).Start(ctx, "MergeSegcoreResults-AppendFieldData")
 		defer span3.End()
-		cursors = make([]int64, len(segmentResults))
-		for _, sel := range selected {
-			retSize += typeutil.AppendFieldData(ret.FieldsData, segmentResults[sel].GetFieldsData(), cursors[sel])
+		for _, selection := range selections {
```

Comment:
this part code is not right, exist a bug.  selection.resultIndex is not correct, because segmentResults is result that from retrieveByoffset in the second parse.
can modify as:
	segmentResOffset := make([]int64, len(segmentResults))
		for _, selection := range selections {
			batchIndex := selection.batchIndex
			retSize += typeutil.AppendFieldData(ret.FieldsData, segmentResults[batchIndex].GetFieldsData(), segmentResOffset[batchIndex])

			// limit retrieve result to avoid oom
			if retSize > maxOutputSize {
				return nil, fmt.Errorf("query results exceed the maxOutputSize Limit %d", maxOutputSize)
			}
			segmentResOffset[batchIndex]++
		}